src/core/domain/userdm/user.go

```diff
@@ -0,0 +1,82 @@
+package userdm
+
+import (
+	"github.com/kazu1029/ddd-menta-sample/src/core/domain/vo"
+	"golang.org/x/xerrors"
+)
+
+type User struct {
+	id                vo.UserID
+	userName          string
+	email             vo.Email
+	password          vo.Password
+	selfIntroduction  string
+	skillIDs          []vo.TagID
+	workExperienceIDs []WorkExperienceID
```

Comment:
この２つはuser_skillとuser_work_experienceを同じユーザという集約内に持っているので、
その場合はユーザエンティティはid参照ではなくオブジェクトを参照しても良いです。
つまり
```suggestion
	skillIDs          []*UserSkill
	workExperienceIDs []*UserWorkExperience
```

この様になります。
エンティティがエンティティを参照してよいのは同じ集約内にある場合のみになります。
理由は集約を跨いでオブジェクトを参照できてしまうと、集約内で他の集約の操作ができてしまい、集約内の知識が他の集約に漏れてしまう可能性があるからです。

Reply:
> エンティティがエンティティを参照してよいのは同じ集約内にある場合のみになります

集約を跨ぐ際はIDなどの識別子を参照するということですね。参考に修正してみます。